pandas/core/indexes/frozen.py

```diff
@@ -68,7 +67,7 @@ def difference(self, other):
         return type(self)(temp)
 
     # TODO: Consider deprecating these in favor of `union` (xref gh-15506)
-    __add__ = __iadd__ = union
+    __add__ = __iadd__ = union  # type: ignore
```

Comment:
The error now is essentially because union is unannotated, plus maybe a few more issues with subclass. In any case we should annotate instead of ignoring, especially since there's no rush on this one (current mypy in CI is still only 0.72)

Reply:
sounds good, ill see if annotating union does the trick

Reply:
looks like it worked for union, but not for _disabled.  removed the "type: ignore"s